fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java

```diff
@@ -98,12 +98,22 @@ public CommandProcessingResult buybackLoanByLoanId(JsonCommand command) {
         LocalDate settlementDate = getSettlementDateFromJson(json);
         ExternalId externalId = getTransferExternalIdFromJson(json);
         validateSettlementDate(settlementDate);
+        validateExternalId(externalId);
```

Comment:
Probably it is just me but I don't get why the external id validation is not required for the sale.

Reply:
No, you are right... it should behave the same.